docs(python): Update stream mode in main tracing files (tracing part 1/3) (#18531)
getsentry/sentry-docs: docs/platforms/python/tracing/instrumentation/custom-instrumentation/index.mdx

            eat_slice(pizza.slices.pop())
```

```python {tabTitle:Stream Mode}
import sentry_sdk

def eat_slice(slice):
    ...

def eat_pizza(pizza):
    with sentry_sdk.traces.start_span(
        name="Eat Pizza",
        attributes={"sentry.op": "task"},

>         # remove the parent span to create service span

Reviewer: ```suggestion
        # set parent span to None to create service span
```

        parent_span=None,
    ):
        while pizza.slices > 0:
            eat_slice(pizza.slices.pop())
```

The [API reference](https://getsentry.github.io/sentry-python/api.html#performance-monitoring) documents `start_transaction` and `start_span`, along with their parameters.

<Alert>

> Note that `sentry_sdk.start_transaction()` (transaction mode) and `sentry_sdk.traces.start_span()` (stream mode) are meant be used as context managers. This ensures that the transaction/service span will be properly set as active and any spans created within will be attached to it.

Reviewer: ```suggestion
Note that `sentry_sdk.start_transaction()` (transaction mode) is meant be used as a context manager. This ensures that the transaction/service span will be properly set as active and any spans created within will be attached to it.

In stream mode, `sentry_sdk.traces.start_span()` can, but doesn't have to be, used as a context manager. If you don't use it as a context manager, make sure to call `span.end()` to finish the span.
```

            eat_slice(pizza.slices.pop())
```

```python {tabTitle:Stream Mode}
import sentry_sdk

@sentry_sdk.traces.trace
def eat_slice(slice):
    ...

def eat_pizza(pizza):
    with sentry_sdk.traces.start_span(
        name="Eat Pizza",
        attributes={"sentry.op": "task"},
        parent_span=None,
    ):
        while pizza.slices > 0:
            eat_slice(pizza.slices.pop())
```

See the [@sentry_sdk.trace decoration section](#sentry_sdktrace-decorator) below for more details.

<Alert level="warning" title="Static & class methods">

> When tracing a static or class method, you **must** add the `@sentry_sdk.trace` decorator **after** the `@staticmethod` or `@classmethod` decorator (i.e., **closer** to the function definition). Otherwise, your function will break! This applies in both transaction mode and stream mode.

Reviewer: ```suggestion
When tracing a static or class method, you **must** add the `@sentry_sdk.trace` (transaction mode)/`@sentry.sdk.traces.trace` (stream mode) decorator **after** the `@staticmethod` or `@classmethod` decorator (i.e., **closer** to the function definition). Otherwise, your function will break! This applies in both transaction mode and stream mode.
```

            span.finish()
```

```python {tabTitle:Stream Mode}
import sentry_sdk

def eat_slice(slice):
    ...

def eat_pizza(pizza):
    with sentry_sdk.traces.start_span(
        name="Eat Pizza",
        attributes={"sentry.op": "task"},
        parent_span=None,
    ):
        while pizza.slices > 0:
            span = sentry_sdk.traces.start_span(name="Eat Slice")
            eat_slice(pizza.slices.pop())

>             span.finish()

Reviewer: `finish` will work too, but it's deprecated.

```suggestion
            span.end()
```

    parent_span.finish()
```

```python {tabTitle:Stream Mode}
import sentry_sdk

def chew():
    ...

def eat_slice(slice):
    parent_span = sentry_sdk.traces.start_span(name="Eat Slice")

    # pass the parent span via `parent_span` to create a child span
    child_span = sentry_sdk.traces.start_span(name="Chew", parent_span=parent_span)
    chew()
    child_span.finish()

>     parent_span.finish()

Reviewer: ```suggestion
    child_span.end()

    parent_span.end()
```

```

In transaction mode, the parameters of `start_span()` and `start_child()` are the same. See the [API reference](https://getsentry.github.io/sentry-python/api.html#sentry_sdk.api.start_span) for more details.

> In stream mode, however, there's no separate `start_child()` method. Instead, pass the parent span explicitly via `parent_span` when starting the child (as shown in the example above).

Reviewer: ```suggestion
In stream mode, however, there's no separate `start_child()` method. Instead, the span will become the child of the currently active span. Alternatively, you can pass the parent span explicitly via `parent_span` when starting the child (as shown in the example above).
```
